nixos/modules/installer/cd-dvd/iso-image.nix

```diff
@@ -186,7 +188,10 @@ let
 
     # Fonts can be loaded?
     # (This font is assumed to always be provided as a fallback by NixOS)
-    if loadfont (hd0)/EFI/boot/unicode.pf2; then
+    if loadfont /EFI/boot/unicode.pf2; then
+      set with_fonts=true
+    fi
+    if [ "\$textmode" != "true" -a "\$with_fonts" == "true" ]; then
```

Comment:
You are sure we need to escape this in nix?

Reply:
Note that, first, this is not shell script, but grub's almost posix shell script look-alike. In case it would be confusing. It is confusing and can be confusing :).

Yes, this is because of how it was implemented. This is in a bash heredoc (yes). Now is not the time to rewrite the whole iso-image derivation, but whew it sure gave me the envy to do so.